atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransactionManager.java

```diff
@@ -42,45 +47,106 @@
 public class SerializableTransactionManager extends SnapshotTransactionManager {
 
     public static class InitializeCheckingWrapper extends AutoDelegate_SerializableTransactionManager {
-        private final SerializableTransactionManager manager;
+        private final SerializableTransactionManager txManager;
         private final Supplier<Boolean> initializationPrerequisite;
+        private final Callback callback;
 
-        public InitializeCheckingWrapper(SerializableTransactionManager manager,
-                Supplier<Boolean> initializationPrerequisite) {
-            this.manager = manager;
+        private volatile boolean callbackDone = false;
+        private Optional<Exception> callbackException = Optional.empty();
+
+        private final ScheduledExecutorService executorService = PTExecutors.newSingleThreadScheduledExecutor(
```

Comment:
This executor isn't shutdown when the TM is closed.  No new tasks will be added but we should still clean up after ourselves.

It is also possible Atlas should try and fewer ThreadPools which are shared by many classes and then we can manage them more easily.